labelbox/data/annotation_types/data/tiled_image.py

```diff
@@ -0,0 +1,525 @@
+        xyz_format = "/{z}/{x}/{y}"
+        if xyz_format not in url:
+            raise ValueError(f"{url} needs to contain {xyz_format}")
+        return url
+
+
+class TiledImageData(BaseData):
+    """ Represents tiled imagery 
+
+    If specified version is 2, converts bounds from [lng,lat] to [lat,lng]
+
+    Requires the following args:
+        tile_layer: TileLayer
+        tile_bounds: TiledBounds
+        zoom_levels: List[int]
+    Optional args:
+        max_native_zoom: int = None
+        tile_size: Optional[int]
+        version: int = 2
+        alternative_layers: List[TileLayer]     
+
+    >>> tiled_image_data = TiledImageData(tile_layer=TileLayer,
+                                  tile_bounds=TiledBounds,
+                                  zoom_levels=[1, 12])
+    """
+    tile_layer: TileLayer
+    tile_bounds: TiledBounds
+    alternative_layers: List[TileLayer] = []
+    zoom_levels: Tuple[int, int]
+    max_native_zoom: Optional[int] = None
+    tile_size: Optional[int] = DEFAULT_TMS_TILE_SIZE
+    version: Optional[int] = 2
+    multithread: bool = True
+
+    def __post_init__(self):
+        if self.max_native_zoom is None:
+            self.max_native_zoom = self.zoom_levels[0]
+
+    def asdict(self):
+        return {
+            "tileLayerUrl": self.tile_layer.url,
+            "bounds": [[
+                self.tile_bounds.bounds[0].x, self.tile_bounds.bounds[0].y
+            ], [self.tile_bounds.bounds[1].x, self.tile_bounds.bounds[1].y]],
+            "minZoom": self.zoom_levels[0],
+            "maxZoom": self.zoom_levels[1],
+            "maxNativeZoom": self.max_native_zoom,
+            "epsg": self.tile_bounds.epsg.name,
+            "tileSize": self.tile_size,
+            "alternativeLayers": [
+                layer.asdict() for layer in self.alternative_layers
+            ],
+            "version": self.version
+        }
+
+    def as_raster_data(self,
+                       zoom: int = 0,
+                       max_tiles: int = 32,
+                       multithread=True) -> RasterData:
+        """Converts the tiled image asset into a RasterData object containing an
+        np.ndarray.
+
+        Uses the minimum zoom provided to render the image.
+        """
+        if self.tile_bounds.epsg == EPSG.SIMPLEPIXEL:
+            xstart, ystart, xend, yend = self._get_simple_image_params(zoom)
+        elif self.tile_bounds.epsg == EPSG.EPSG4326:
+            xstart, ystart, xend, yend = self._get_3857_image_params(zoom)
+        elif self.tile_bounds.epsg == EPSG.EPSG3857:
+            #transform to 4326
+            transformer = EPSGTransformer.create_geo_to_geo_transformer(
+                EPSG.EPSG3857, EPSG.EPSG4326)
+            self.tile_bounds.bounds = [
+                transformer(self.tile_bounds.bounds[0]),
+                transformer(self.tile_bounds.bounds[1])
+            ]
+            xstart, ystart, xend, yend = self._get_3857_image_params(zoom)
```

Comment:
I would change _get_3857_image_params to accept bounds instead of using self.bounds. That way you won't have to project, overwrite the variable, and then reproject back.